### ISSUE TITLE
Print setup disabled. G-Code file can not be modified

### ISSUE THREAD

**Issue by omdathetkan** [Ghostkeeper/SVGToolpathReader#3]
I am not sure if this is a cure bug or restriction, but when you open a SVG file it will switch to the preview screen which shows "Print setup disabled. G-code file can not be modified":
![image](https://user-images.githubusercontent.com/5098411/77476882-bea6db80-6e1b-11ea-86fb-ed47b23dd397.png)

Also when navigating back to the prepare panel this message remains. If I understand correctly this plugin uses various cura profile settings. Right now the only way to get the settings back is by first opening a random STL file, and then re-opening the SVG.

**Comment by omdathetkan** [comment]
Additionally (once this is resolved) it might be useful if it configures setting visibility to show only what is supported by the plugin. I think this should be possible based on this plugin:
https://github.com/fieldOfView/Cura-SettingVisibilityProfileCreator/blob/3.5/SettingVisibilityProfileCreator.py

Would it be possible to create custom settings for the plugin (to replace the settings dialog every time a file is opened?)

**Comment by Ghostkeeper** [comment]
This is sort of inherent to the way in which this plug-in works. It doesn't create a mesh that is then sliced by CuraEngine. It generates g-code. Essentially it loads in an SVG file and tells Cura that it contains g-code. It generates that g-code on the fly without passing through CuraEngine.

When Cura loads in a g-code file it displays that message. And indeed the message is correct as it can't modify the g-code any more after it's been generated. Since generating happens upon loading the SVG file, it'd need to re-load the model.

It's a bit of an architectural issue. To fix it I'd essentially need to make the scene contain the SVG file in some way, and actually "slice" that. To slice that I'd have to replace CuraEngine in the slice sequence, which has never been done before as far as I know. Or I'd need to let CuraEngine slice the scene normally, but that defeats a large part of the purpose of this plug-in to produce paths along the outlines of the SVGs regardless of their shapes. E.g. you can't produce single lines any more, everything has to be closed loops. That sort of thing.

In short, it'd be pretty hard to fix this.

**Comment by omdathetkan** [comment]
Hm yeah I understand that, but wouldn't it be possible to somehow trick Cura to bring back the settings dialog without first having to open an STL file? Can you unload the g-code or something?

**Comment by omdathetkan** [comment]
Ahhhhhhhhhh..... I found there is a "Clear Build Plate" command. This does the trick. Maybe good to highlight this in the readme?

**Comment by Ghostkeeper** [comment]
I've added a note under the "Usage" header to clarify this a bit.
